packages/cubejs-query-orchestrator/src/orchestrator/RedisFactory.ts

```diff
@@ -64,10 +22,5 @@ export function createRedisClient(url: string, opts: ClientOpts = {}) {
     options.password = process.env.REDIS_PASSWORD;
   }
 
-  return decorateRedisClient(
-    redis.createClient({
-      ...options,
-      ...opts,
-    })
-  );
+  return Promise.resolve(new Redis(options));
```

Comment:
I think it's not needed to wrap `new Redis` with `Promise.resolve` because `createRedisClient` function already declared as `async` and it will use `AsyncFunction` type which always return Promise on call.

Maybe I miss something.

Reply:
No, it is not needed. The async will add it